### DISCUSSION TITLE
How could I migrating an existing project and only use the typed routes feature?

### DISCUSSION THREAD

**Discussion by s3xysteak** [posva/unplugin-vue-router#526] [Questions and Help]
I wanna migrate an existing project, and do not change any behaviors. I just want to have types linting for `vue-router`. I came here from `vue-router` so I thought it just provide typed routes feature for vue-router. But I found more features like File-based Routing... I'm worried that this will cause breaking changes to the project so I open this discussion.

Besides that, how about adding a solution [in the document of vue-route](https://router.vuejs.org/guide/advanced/typed-routes.html) as well?

**Comment by PindaPixel** [comment]
Types are generated from the file system and not the routes array you pass to createRouter, so sadly this doesn't seem possible at the moment.

On the other hand, you could maybe use the definePage macro on all your pages instead.
